internal/oci/runtime_vm.go

```diff
@@ -109,13 +111,32 @@ func (r *runtimeVM) CreateContainer(c *Container, cgroupParent string) (err erro
 	containerIO.AddOutput("logfile", f, f)
 	containerIO.Pipe()
 
+	r.Lock()
 	r.ctrs[c.ID()] = containerInfo{
 		cio: containerIO,
 	}
+	r.Unlock()
 
 	defer func() {
 		if err != nil {
+			logrus.WithError(err).Warnf("Cleaning up container %v", c.ID())
+
+			if cleanupErr := containerIO.Close(); cleanupErr != nil {
+				logrus.WithError(cleanupErr).Infof("Error closing the container %v IO", c.ID())
+			}
+
+			if cleanupErr := r.remove(r.ctx, c.ID(), ""); cleanupErr != nil {
+				logrus.WithError(cleanupErr).Infof("Error removing the container %v from the background context", c.ID())
+			}
+
+			_, cleanupErr := r.task.Shutdown(r.ctx, &task.ShutdownRequest{ID: c.ID()})
+			if cleanupErr != nil && !errors.Is(cleanupErr, ttrpc.ErrClosed) {
+				logrus.WithError(cleanupErr).Infof("Error error shutting down %v", c.ID())
```

Comment:
"Error error"?  I think just "Error" is sufficient.

Reply:
```suggestion
				logrus.WithError(cleanupErr).Infof("Error shutting down container %v", c.ID())
```

Reply:
That was a typo, thanks! :-)